index.html

```diff
@@ -204,55 +205,2367 @@ <h3>
+            </li>
+          </ol>
+        </dd>
+        <dt>
+          <dfn>collections</dfn> attribute
+        </dt>
+        <dd>
+          <p>
+            A sequence of {{HIDCollectionInfo}} representing the [=top-level
+            collections=] in the [=report descriptor=].
+          </p>
+          <p>
+            The {{HIDDevice/collections}} getter steps are:
+          </p>
+          <ol>
+            <li>Return [=this=].{{HIDDevice/[[collections]]}}.
+            </li>
+          </ol>
+          <div class="note">
+            <dfn>Top-level collections</dfn> group items that serve similar
+            purposes within the device. Applications look at the [=HID usage=]
+            applied to [=top-level collections=] to identify devices. An
+            application may not recognize a device if it does not follow the
+            conventions for a particular type of device. Refer to the <a href=
+            "https://usb.org/document-library/hid-usage-tables-122">HID Usage
+            Tables</a> document for more information about the conventional
+            behavior for standard HID devices.
+          </div>
+        </dd>
+      </dl>
+      <p>
+        When an input report is received from a |device:HIDDevice|, run the
+        following steps:
+      </p>
+      <ol>
+        <li>If the input report is a [=blocked report=], abort these steps.
+        </li>
+        <li>Let |reportId:octet| be the report ID for this report, or 0 if the
+        [=HID interface=] does not [=use report IDs=].
+        </li>
+        <li>Let |data:DataView| be a {{DataView}} created over the [=byte
+        sequence=] representing the input report. If the [=HID interface=]
+        [=uses report IDs=], the [=byte sequence=] MUST NOT include the report
+        ID byte.
+        </li>
+        <li>[=Fire an event=] named {{inputreport}} at |device| using
+        {{HIDInputReportEvent}} with its {{HIDInputReportEvent/device}}
+        attribute initialized to |device|, its {{HIDInputReportEvent/reportId}}
+        attribute initialized to |reportId|, and its
+        {{HIDInputReportEvent/data}} attribute initialized to |data|.
+        </li>
+      </ol>
+      <section>
+        <h3>
+          <dfn>open()</dfn> method
+        </h3>
+        <p>
+          The {{HIDDevice/open()}} method steps are:
+        </p>
+        <ol>
+          <li>Let |promise:Promise| be [=a new promise=].
+          </li>
+          <li>If [=this=].{{HIDDevice/[[state]]}} is not `"closed"`, then
+          [=reject=] |promise| with a "{{InvalidStateError}}" {{DOMException}}
+          and return |promise|.
+          </li>
+          <li>Set [=this=].{{HIDDevice/[[state]]}} to `"opening"`.
+          </li>
+          <li>Set [=this=].{{HIDDevice/[[pendingOpenPromise]]}} to |promise|.
+          </li>
+          <li>Perform the following steps [=in parallel=]:
+            <ol>
+              <li>If [=this=].{{HIDDevice/[[pendingOpenPromise]]}} is not
```

Comment:
I don't think this is necessary. `close()` should throw if called when the state is "opening".